memorygame.js

```diff
@@ -0,0 +1,123 @@
+//Memory Game
+// variables
+var startPlayButton = false;
+var firstImage;
+var secondImage;
+var score = 0;
+var clickCount = 0;
+var firstImageId;
+var defaultImage = "./img/startImage.jpg"
+var compareCompleted;
+var activeImages = [];
+var imagePosition = [];
+document.getElementById("startPlay").addEventListener("click",startGame);
+// document.getElementById("startPlay").addEventListener("click",setTimeout(hideImage,2000)); // not working on click
+// document.getElementById("startPlay").onclick = setTimeout(hideImage,2000); //not working on click
+// document.getElementById("startPlay").onclick = hideImage(); //not working on click
+function startGame() {
+	hideImage();
+	variableValueReset();
+	reorderImages();
+	displayResult();
+};
+function clickedImg(id) {
+	if ((compareCompleted)&&(startPlayButton)&&(activeImages[id])) {
+		compareCompleted = false;
+		activeImages[id] = false;
+		showImage(id);
+		compare(id);
+	};
+};
+function hideImage() {
+	for (var i = 0; i < document.images.length; i++) {
+		document.images[i].src = defaultImage;
+		document.images[i].style.border = "5px solid grey";
+	};
+};
+// for (var i = 0; i <= document.images.length -1; i++) {
+// 	document.images[i].addEventListener("click",showImage(i));
+// };
+function showImage(imageId) {
+	document.images[parseInt(imageId)].src = imagePosition[parseInt(imageId)];
+	document.images[parseInt(imageId)].style.border = "5px solid red";
```

Comment:
Cache variables to avoid over processing

Example
```js
var index = parseInt(imageId);

document.images[index].src = imagePosition[index];
document.images[index].style.border = "5px solid red";
```